File: src/main/java/com/example/interviewlandbackend/dto/response/QuestionDto.java
package com.example.interviewlandbackend.dto.response;

public class QuestionDto {

    private int questionId;

    private String title;

    private String description;

    private String sourceCode;

    private int sectionId;

    public QuestionDto(int questionId, String title,
                       String description, String sourceCode, int sectionId) {
        this.questionId = questionId;
        this.title = title;
        this.description = description;
        this.sourceCode = sourceCode;
        this.sectionId = sectionId;
    }
}

File: src/main/java/com/example/interviewlandbackend/dto/response/QuestionDto.kt
package com.example.interviewlandbackend.dto.response

import com.example.interviewlandbackend.model.Question
import java.util.*

data class QuestionDto @JvmOverloads constructor(

    val questionId: Int?,
    val questionTitle: String,
    val description: String,
    val sourceCode: String,
    val sectionId: Int?,
    val createdDate: Date?

){

    companion object{

        @JvmStatic fun convert(question: Question) : QuestionDto{
            return QuestionDto(
                question.id,
                question.questionTitle,
                question.description,
                question.sourceCode,
                question.section.id,
                question.date
            )
        }

    }

}
